### ISSUE TITLE
Plugin System

### ISSUE THREAD

**Issue by rgbkrk** [RobCoIndustries/pipboy#13]
It would be really great if people could register plugins that create their own view or just run some helper utilities. I'm imagining we would hand them either a reference to the `subject` or some Object that had `subject` and `database`.

**Comment by luckydonald** [comment]
Should this include adding own tabs?

**Comment by rgbkrk** [comment]
Yeah, I'm kind of thinking that we assume a plugin author will provide us a `React.Component` and that the name we display is some class property (e.g. using `displayName`).

**Comment by kitten** [comment]
@rgbkrk I think that's a good idea. A system that allows the author to tap into the connection all the time is probably a lot more powerful though. We have to come up with something different for that I guess.

**Comment by luckydonald** [comment]
Maybe we should think about what plugins use, what are they supposed to do.
Do they enhance the UI by something clickable?
Do they just talk with the game?
Do the need settings?

_Example ideas:_
Some non-gui mod could be an autoinject steampak mod.
(I saw this somewhere here)    

Some gui thing: an radar for your quests.
    [radar.png](http://exchangeserverpro.com/wp-content/uploads/2009/12/radar.jpg)

**Comment by kitten** [comment]
@luckydonald Yea, definitely make sense to list these requirements. Imho, plugins should be able to:
- Access the connection subject (receive and send payloads)
- Add one or more new panes
- Add settings to a kind of plugin page

I think all of these requirements can be easily fulfilled. I think an easy solution would be:

A plugin is an npm (like) package. The package's main file is imported inside the pipboy app. Now either the pipboy app injects a module / dependency that exposes our plugin API (quite complex) or it just imports a function that each plugin exports.

The API might expose certain methods that can add components as panes, or add settings. All background logic then actually doesn't have to be implemented explicitly, but would be just a side effect
